subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestOptionsIntegrationSpec.groovy

```diff
@@ -405,6 +407,179 @@ public class SomeIntegTestClass {
     }
     // endregion stand-alone test tasks
 
+    @Issue("https://github.com/gradle/gradle/issues/24331")
+    def "options can be set with framework before java plugin is applied"() {
+        given:
+        settingsFile << "rootProject.name = 'Sample'"
+        buildFile.text = """
+            tasks.withType(Test).configureEach {
+                useJUnitPlatform() {
+                    excludeTags = ["Slow"]
+                }
+                doFirst {
+                    assert options.excludeTags.contains("Slow")
+                }
+            }
+
+            apply plugin: 'java'
+
+            ${mavenCentralRepository()}
+            dependencies {
+                testImplementation 'org.junit.jupiter:junit-jupiter:${JUnitCoverage.LATEST_JUPITER_VERSION}'
+            }
+        """.stripIndent()
+
+        expect:
+        succeeds("check", "--stacktrace")
```

Comment:
❔ What's the need for `--stacktrace` in all these tests?

Reply:
Bah, that helped with diagnosing incorrect behaviors.  I just forgot to remove them.